### PR TITLE
Fix the problem of using option to append Dataset public option

### PR DESCRIPTION
**Pull request by lgy1027**
Fix error when using sharedOptions to mount multiple mounts

image

### CODE REVIEW

File: pkg/ddc/alluxio/ufs_internal.go [hunk 1]
```diff
@@ -372,8 +372,8 @@ func (e *AlluxioEngine) genUFSMountOptions(m datav1alpha1.Mount, SharedOptions m
 
 	// initialize alluxio mount options
 	mOptions := map[string]string{}
-	if len(SharedOptions) > 0 {
-		mOptions = SharedOptions
+	for k, v := range SharedOptions {
```
Comment (cheyang) [inline pkg/ddc/alluxio/ufs_internal.go:375]: Could you also add unit test for the fix? thanks very much!

Comment (zhang-x-z) [reply pkg/ddc/alluxio/ufs_internal.go:375]: I have added unit test: generating UFS mount options multiple times via functions `genUFSMountOptions` in `AlluxioEngine`

**Review by cheyang — APPROVED** [review]
/lgtm
/approve

**Comment by cheyang** [comment]
Thanks very much for your contribution, @lgy1027 @zhang-x-z

**Pull request merged**